Tribler/Core/Modules/restapi/downloads_endpoint.py

```diff
@@ -375,6 +381,14 @@ def render_PATCH(self, request):
 
         parameters = http.parse_qs(request.content.read(), 1)
 
+        if 'selected_files[]' in parameters:
+            selected_files_list = parameters['selected_files[]']
+            if len(selected_files_list) == 0:
+                request.setResponseCode(http.BAD_REQUEST)
+                return json.dumps({"error": "a download needs at least one selected file"})
```

Comment:
Unless a download with no included files breaks some internal Tribler expectations, this check should be in GUI side, not in API handler.

When selected_files_list consts of just invalid (non-existing) file names, it works like empty included list, placing all files into `.unwanted`. Although it may seem useless for user, it should not be probably banned here.